### DISCUSSION TITLE
Why are presets not unified across editors?

### DISCUSSION THREAD

**Discussion by Tomagori** [openstreetmap/iD#12202] [Q&A] [answered]
So, this is less of an issue with iD specifically, but something that iD could easily take the lead on, considering its importance as an editor.

OSM Wiki notes ["Presets have not been standardized across editors"](https://wiki.openstreetmap.org/wiki/Preset). It seems a first attempt at doing this was considered [some years ago](https://github.com/osmlab/editor-presets), but nothing ever came of it.

My question is: Why not? 

There are a ton of obvious benefits to having unified presets. Data quality of OSM, ease of use for editors, communicability of editing practices, accuracy of rendering by editors would all increase massively if presets (and therefore tagging schemes) were standardized more. The wiki, in integrating that standard preset scheme, could offer even better documentation of those tagging schemes. Changes in tagging could easily propagate across geography and time, past tagging could be understood better as a result of previous unified standards, editing and adding data would be simplified considerably and the barrier of entry for new editors (and editor solutions) would be lowered as well.

I realise this is nothing that iD can change on its own. No change in iD's code is going to bring about a unified preset solution. But why haven't there been more attempts at integrating the presets JOSM, Potlach, iD and other editing solutions use? We see some convergence; StreetComplete uses iD's presets, Vespucci uses JOSM's. But still, a (IMO highly unnecessary) rift remains, only getting worse with time as more editors, editing solutions and tagging discrepancies emerge.

EDIT:
JOSM ticket: https://josm.openstreetmap.de/ticket/20995
Vespucci issue: https://github.com/MarcusWolschon/osmeditor4android/issues/1412

**Comment by lectrician1** [comment]
From what I see in the JOSM repo, all of the presets are XML-based. iD uses a JSON-based solution and it is all located in a separate repository (https://github.com/openstreetmap/id-tagging-schema).

iD's solution seems better because it is more organized and easier to interpret than what JOSM uses. I'd forward this issue to JOSM's trac and Vespucci's repo ask them why they don't switch to the tagging schema repository.

**Comment by matkoniecz** [comment, marked as answer]
See also comment in https://josm.openstreetmap.de/ticket/20995#comment:5

> Vespucci uses JOSM's

Not true. Vespucci is using fork of JOSM presets - see https://github.com/simonpoole/beautified-JOSM-preset

Also, this issue should be rather in https://github.com/openstreetmap/id-tagging-schema anyway.

> My question is: Why not?

Historical answer: because iD and Vespucci preferred to start own presets rather than reuse JOSM one, possibly expanding it.

Why it is not changing?

Partially because people have different ideas how various features should be tagged, described and which tags should be promoted (see for example `highway=track` discussions). So, unifying presets would make decisions made much more powerful.

Presets are very important, not taking so much effort to maintain existing ones - so if an editor has working presets there is no very big benefit from starting to rely on external one.

In addition different editors have different needs - for example see different styles in icons between iD and JOSM. Shared preset would require both, so no actual reduction in duplication would be achieved here.

Note that in general iD preset is very tightly coupled to how it is used specifically in iD.

Still, some work may be done (like detecting which features are missing in given presets and present in others), and if someone would be interested in working on that - it would be likely useful.

Similarly reducing other incompatibilities may help.

> From what I see in the JOSM repo, all of the presets are XML-based. iD uses a JSON-based solution

This is not a real problem and you are missing issues actually present. If this would be a sole problem then it would be trivially solvable within hours.

> I'd forward this issue to JOSM's trac and Vespucci's repo ask them why they don't switch to the tagging schema repository.

There would be a noticeable loss of functionality, JOSM and Vespucci has presets for many things missing in iD and many parts fitting specifically iD would be not fitting Vespucci and JOSM.

**Comment by lectrician1** [comment]
> There would be a noticeable loss of functionality, JOSM and Vespucci has presets for many things missing in iD and many parts fitting specifically iD would be not fitting Vespucci and JOSM.

Can you elaborate specifically? (I'm curious and I don't know)

**Comment by matkoniecz** [comment]
for start

> In addition different editors have different needs - for example see different styles in icons between iD and JOSM. Shared preset would require both, so no actual reduction in duplication would be achieved here.

Also, compare list of features in Vespucci presets with what is in iD presets. iD misses some.

**Comment by TheAdventurer64** [comment]
> > There would be a noticeable loss of functionality, JOSM and Vespucci has presets for many things missing in iD and many parts fitting specifically iD would be not fitting Vespucci and JOSM.
> 
> Can you elaborate specifically? (I'm curious and I don't know)

Well, JOSM seems to only use single tags and using the wiki version of them for efficiency, whereas iD has a lot more presets utilizing several tags and uses different tagging names depending on the area (Example: The "American Restaurant" preset in iD is `amenity=restaurant` + `cuisine=restaurant`. JOSM does not have this preset for grouping purposes and simplicity; In iD, `amenity=fuel` is shown as "Gas Station", whereas JOSM uses simply "Fuel".)

**Comment by matkoniecz** [comment]
@TheAdventurer64 This is not a problematic part as far as I see
